python_modules/dagster/dagster/core/definitions/context.py

```diff
@@ -26,6 +26,27 @@ class PipelineContextDefinition(object):
 
     The ``PipelineContextDefinition`` is passed to the ``PipelineDefinition`` in
     a dictionary keyed by its name so the name is not present in this object.
+    
+    Args:
+        context_fn (Callable):
+            Signature of context_fn:
+            (pipeline: PipelineDefintion, config_value: Any) => ExecutionContext
+
+            Returns *or* yields an ExecutionContext.
+
+            If it yields a context, the code after the yield executes after pipeline
+            completion, just like a python context manager.
+
+            Environment-specific resources should be placed in the "resources" argument
+            to an execution context. This argument can be *anything* and it is made
+            avaiable to every solid in the pipeline. A typical pattern is to have this
```

Comment:
So the resources thing in the ExecutionContext is mostly for advanced cases. We should even consider disallowing it. The first class way of doing this is to use ResourceDefinitions. If resource definitions are present then setting the resources property in the context_fn is an error.